UPGRADES.md

```diff
@@ -0,0 +1,204 @@
+Upgrading is a multi step process, we need to schedule first, then execute. Having an intermediary file helps the auditability
+of the process.
+
+Remember to use CREATE3 for proxies
+
+Example:
+
+```solidity
+contract DeployerV1_2 is JsonDeploymentHandler, BroadcastManager, UpgradedImplHelper {
+    ///
+
+    string constant PREV_VERSION = "v1.1.1";
+    string constant PROPOSAL_VERSION = "v1.2.0";
+
+
+    constructor() JsonDeploymentHandler("main") {
+        create3Deployer = ICreate3Deployer(CREATE3_DEPLOYER);
+    }
+
+    function run() public virtual {
+        _readDeployment(PREV_VERSION); // JsonDeploymentHandler.s.sol
+        // Load existing contracts
+        protocolAccessManager = AccessManager(_readAddress("ProtocolAccessManager"));
+        /// ...
+
+        _beginBroadcast(); // BroadcastManager.s.sol
+
+        UpgradeProposal[] memory proposals = deploy();
+        _writeUpgradeProposals(PREV_VERSION, PROPOSAL_VERSION, proposals); // JsonDeploymentHandler.s.sol
+
+        _endBroadcast(); // BroadcastManager.s.sol
+    }
+
+    function deploy() public returns (UpgradeProposal[] memory) {
+        string memory contractKey;
+        address impl;
+
+        // Deploy new contracts
+        _predeploy("RoyaltyPolicyLRP");
+        impl = address(new RoyaltyPolicyLRP(address(royaltyModule)));
+        royaltyPolicyLRP = RoyaltyPolicyLRP(
+            TestProxyHelper.deployUUPSProxy(
+                create3Deployer,
+                _getSalt(type(RoyaltyPolicyLRP).name),
+                impl,
+                abi.encodeCall(RoyaltyPolicyLRP.initialize, address(protocolAccessManager))
+            )
+        );
+        require(
+            _getDeployedAddress(type(RoyaltyPolicyLRP).name) == address(royaltyPolicyLRP),
+            "Deploy: Royalty Policy Address Mismatch"
+        );
+        require(_loadProxyImpl(address(royaltyPolicyLRP)) == impl, "RoyaltyPolicyLRP Proxy Implementation Mismatch");
+        impl = address(0);
+        _postdeploy("RoyaltyPolicyLRP", address(royaltyPolicyLRP));
+        
+        //...
+
+        // Deploy new implementations
+        contractKey = "LicenseToken";
+        _predeploy(contractKey);
+        impl = address(new LicenseToken(licensingModule, disputeModule));
+        upgradeProposals.push(UpgradeProposal({ key: contractKey, proxy: address(licenseToken), newImpl: impl }));
+        impl = address(0);
+
+        //...
+
+        _logUpgradeProposals();
+        
+        return upgradeProposals;
+    }
+}
+```
+
+For `IPRoyaltyVault`, set as proxy the address of the contract that is `IVaultController`
+
+Output will look something like:
+
+`deploy-out/upgrade-v1.1.1-to-v1.2.0-1513.json`
+```
+{
+  "main": {
+    "GroupingModule-NewImpl": "0xa1A9b2cBb4fFEeF7226Eaee9A5b71007bDCa721F",
+    "GroupingModule-Proxy": "0xeD1eF5749468B1805952757F53aB4C9037cD3ed6",
+    // ...
+  }
+}
+```
+
+## 4. Write contracts inheriting UpgradeExecutor
```

Comment:
Could we add the sample command line to execute the upgrade script or added to Make file?

Reply:
Added a sample, since every upgrade has custom forge scripts. The contracts included in the PR are abstract